### ISSUE TITLE
Parenthesis in string causes processing error

### ISSUE THREAD

**Issue by erick166** [jeremiah-c-leary/vhdl-style-guide#1531]
**Environment**
VHDL Style Guide (VSG) version: 3.35.0
Windows 11

**Describe the bug**
VSG fails to process the file and returns the following error:
```
Error while processing C:/uvvm/uvvm_util/src/string_methods_pkg.vhd:
Error: Unexpected token detected while parsing procedure_call @ Line 760, Column 73 in file C:/uvvm/uvvm_util/src/string_methods_pkg.vhd
       Expecting : )
       Found     : ;
```

**To Reproduce**
1. Checkout the latest UVVM version
2. Run VSG on the file uvvm/uvvm_util/src/string_methods_pkg.vhd
(Same issue happens with the files func_cov_pkg.vhd and rand_pkg.vhd)

**Expected behavior**
The error happens when using a parenthesis encompassed by quotes denoting a string, which apparently is detected as a normal parenthesis which is matched with another parenthesis.

**Comment by JHertz5** [comment]
I've narrowed the problem down to this function:
```python
def filter_character_literal_candidates(lLiterals):
    lReturn = []
    for iIndex, lLiteral in enumerate(lLiterals[0:-1]):
        lNextLiteral = lLiterals[iIndex + 1]
        lPreviousLiteral = lLiterals[iIndex - 1]
        if lLiteral[1] == lNextLiteral[0] and lLiteral[0] == lPreviousLiteral[1]: # this line is the problem. Method assumes that first pair is a valid literal.
            continue
        lReturn.append(lLiteral)
    lReturn.append(lLiterals[-1])
    return lReturn
```
The function uses a simple method for skipping character literal candidates that share quotes with other character literal candidates. This works great for some cases, e.g.
```vhdl
my_function('1','(','i');
```
but it assumes that the character literals will always come in pairs, which is not a valid assumption when qualified expressions come into the picture, e.g.
```vhdl
my_function(std_logic'('1'),'(','i');
```
It should be pretty easy to improve this method, I'll have a go.

**Comment by JHertz5** [comment]
Hi @erick166. I've successfully solved the secondary problem that I mentioned, with PR #1534. With the two sets of changes combined, I'm able to successfully process the UVVM file. Once you're happy with the changes for this issue, we can merge the branch and close this issue.

**Comment by erick166** [comment]
Hi! I've managed to test https://github.com/jeremiah-c-leary/vhdl-style-guide/pull/1532 and it solves the issue, so you can go ahead and merge the changes, thanks for the help!

**Comment by JHertz5** [comment]
@erick166 Fantastic, thanks! I'll get that merged in.
